### PR TITLE
refactor!: managers

### PR DESCRIPTION
**Pull request by vishwa2710**
This MR refactors the various Managers by creating a Parent "BaseManager" class.
It is a breaking change as some of the static methods no longer exist, however all the functionality is maintained.

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation in the README for better user guidance.
	- Added detailed instructions for using Docker with JupyterLab.
	- Expanded sections on data management and installation processes.

- **Bug Fixes**
	- Removed outdated tests related to default mode and repository management, streamlining the testing process.

- **Refactor**
	- Simplified the `Manager` class structure by removing unnecessary methods and focusing on core functionalities.
	- Updated the handling of local repository management and operational modes.
	- Transitioned from `<experimental/filesystem>` to `<filesystem>` for improved stability.

- **Tests**
	- Updated test cases to reflect changes in the `Manager` class, ensuring robust validation of functionalities.
	- Added new tests for edge cases in manual mode and streamlined existing tests for clarity.

### CODE REVIEW

File: test/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.test.cpp [hunk 1]
```diff
@@ -51,8 +51,7 @@ TEST(OpenSpaceToolkit_Physics_Environment_Gravitational_Earth, Constructor)
         EXPECT_NO_THROW(EarthGravitationalModel earthGravitationalModel(EarthGravitationalModel::Type::EGM96));
         EXPECT_NO_THROW(EarthGravitationalModel earthGravitationalModel(EarthGravitationalModel::Type::EGM2008));
 
-        EarthGravitationalModelManager::Get().setLocalRepository(EarthGravitationalModelManager::DefaultLocalRepository(
-        ));
+        EarthGravitationalModelManager::Get().reset();
```
Comment (kyle-cochran) [inline test/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.test.cpp:54]: Re: above

Perhaps it's more maintainable to add a `reset()` to the test setup instead of relying on individual tests to clean up after themselves.

Not relevant for this MR though so feel free to close.

File: bindings/python/test/environment/magnetic/earth/test_manager.py [hunk 2]
```diff
@@ -19,8 +14,6 @@
 def manager() -> EarthMagneticModelManager:
     manager = EarthMagneticModelManager.get()
 
-    manager.set_local_repository(EarthMagneticModelManager.default_local_repository())
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply bindings/python/test/environment/magnetic/earth/test_manager.py:22]: So the reason I removed that is because when you instantiate the manager, it by default sets it to the default local repository, so it felt a bit redundant.

File: include/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Manager.hpp [hunk 3]
```diff
@@ -167,74 +135,28 @@ class Manager
...
-
     CSSISpaceWeather CSSISpaceWeather_;
 
-    mutable std::mutex mutex_;
-
-    Manager(const Manager::Mode& aMode = Manager::DefaultMode());
+    Manager();
 
-    bool isLocalRepositoryLocked() const;
+    const CSSISpaceWeather* accessCSSISpaceWeatherAt_(const Instant& anInstant) const;
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Manager.hpp:151]: They are indeed. I haven't changed any of the interfaces, just renamed private methods with an underscore which is something we need to do across the repos.
the public facing `get*` methods are still available 👍

File: test/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.test.cpp (hunk 1 above)
Comment (vishwa2710) [reply test/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.test.cpp:54]: Agreed. Since this is the only test case that is using the manager I thought it was fine to just reset within the function. But great point!

File: src/OpenSpaceToolkit/Physics/Coordinate/Frame/Provider/IERS/Manager.cpp [hunk 4]
```diff
@@ -287,7 +253,7 @@ File Manager::fetchLatestFinals2000A() const
 
 void Manager::reset()
 {
-    std::lock_guard<std::mutex> lock {mutex_};
+    BaseManager::reset();
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Physics/Coordinate/Frame/Provider/IERS/Manager.cpp:256]: Base manager handles the mutex

File: src/OpenSpaceToolkit/Physics/Manager.cpp [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+{
+    std::lock_guard<std::mutex> lock {mutex_};
+
+    return localRepository_;
+}
+
+Duration Manager::getLocalRepositoryLockTimeout() const
+{
+    return localRepositoryLockTimeout_;
+}
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Physics/Manager.cpp:49]: Yes can you create unit tests?

File: bindings/python/src/OpenSpaceToolkitPhysicsPy/Coordinate/Frame/Provider/IERS/Manager.cpp [hunk 6]
```diff
@@ -2,15 +2,18 @@
 
 #include <OpenSpaceToolkit/Physics/Coordinate/Frame/Provider/IERS/Manager.hpp>
 
+#include <OpenSpaceToolkit/Physics/Manager.cpp>
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply bindings/python/src/OpenSpaceToolkitPhysicsPy/Coordinate/Frame/Provider/IERS/Manager.cpp:5]: The bindings classes import .cpp files.

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Provider/IERS/Manager.test.cpp [hunk 7]
```diff
@@ -521,65 +521,4 @@ TEST_F(OpenSpaceToolkit_Physics_Coordinate_Frame_Provider_IERS_Manager, Get)
...
-    }
-}
-
-TEST_F(OpenSpaceToolkit_Physics_Coordinate_Frame_Provider_IERS_Manager, DefaultLocalRepositoryLockTimeout)
-{
-    {
-        EXPECT_EQ(Duration::Seconds(60.0), Manager::DefaultLocalRepositoryLockTimeout());
-    }
-}
+}
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Provider/IERS/Manager.test.cpp:524]:
```suggestion
}

```

File: include/OpenSpaceToolkit/Physics/Manager.hpp [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+    /// @brief              Set local repository
+    ///
+    /// @param              [in] aDirectory A repository directory
+
+    void setLocalRepository(const Directory& aDirectory);
+
+    /// @brief                  Reset the manager.
+    ///
+    ///                         Unload the manifest file and forget manifest age.
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Manager.hpp:95]: Did you intend to have a mention of the manifest file here? Or did you just mean to mention the files in the local repository?

File: include/OpenSpaceToolkit/Physics/Manager.hpp [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+    /// @brief              Get local repository
+    ///
+    /// @return             Local repository
+
+    Directory getLocalRepository() const;
+
+    /// @brief              Get local repository timeout
+    ///
+    /// @return             Local repository timeout
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Manager.hpp:77]: Can you add a small description on what the lock timeout is for? Either here and/or in the README?

**Review by apaletta3 — APPROVED** [review]
Epic stuff man!! A couple nits, but overall looks awesome! Will be much easier to maintain in the future

File: include/OpenSpaceToolkit/Physics/Manager.hpp (hunk 8 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Physics/Manager.hpp:95]: I think this is bad copy pasta. Good catch.
